src/vs/workbench/parts/stats/node/workspaceStats.ts

```diff
@@ -343,36 +399,90 @@ export class WorkspaceStats implements IWorkbenchContribution {
-									if ('react-native' === module) {
-										if (packageJsonContents['dependencies'][module]) {
-											tags['workspace.reactNative'] = true;
-										}
-									} else {
-										if (packageJsonContents['dependencies'][module]) {
-											tags['workspace.npm.' + module] = true;
-										}
-									}
-								}
-							}
-						}
-						catch (e) {
-							// Ignore errors when resolving file or parsing file contents
-						}
+			function getFilePromises(filename, fileService, contentHandler): TPromise<void>[] {
+				return !nameSet.has(filename) ? [] : folders.map(workspaceUri => {
+					const uri = workspaceUri.with({ path: `${workspaceUri.path !== '/' ? workspaceUri.path : ''}/${filename}` });
+					return fileService.resolveFile(uri).then(() => {
+						return fileService.resolveContent(uri, { acceptTextOnly: true }).then(contentHandler);
+					}, err => {
+						// Ignore missing file
 					});
-				}, err => {
-					// Ignore missing file
 				});
+			}
+
+			const requirementsTxtPromises = getFilePromises('requirements.txt', this.fileService, content => {
+				try {
+					let dependencies: string[] = content.value.split('\n');
+					if (!tags['workspace.py.any-azure']) {
+						tags['workspace.py.any-azure'] = WorkspaceStats.searchArray(dependencies, /azure/i);
+					}
+
+					for (let dependency of dependencies) {
+						// Dependencies in requirements.txt can have 3 formats: `foo==3.1, foo>=3.1, foo`
+						const format1 = dependency.split('==')[0];
+						const format2 = dependency.split('>=')[0];
+						let packageName = format1.length > format2.length ? format1.trim() : format2.trim();
+						if (PyModulesToLookFor.indexOf(packageName) > -1) {
+							tags['workspace.py.' + packageName] = true;
+						}
+						// cognitive services has a lot of tiny packages. eg. 'azure-cognitiveservices-search-autosuggest'
+						if (packageName.indexOf('azure-cognitiveservices') > -1) {
+							tags['workspace.py.cognitiveservices'] = true;
+						}
+					}
+				} catch (e) {
+					// Ignore errors when resolving file or parsing file contents
+				}
 			});
 
-			return TPromise.join(packageJsonPromises).then(() => tags);
+			const pipfilePromises = getFilePromises('pipfile', this.fileService, content => {
+				try {
+					let dependencies: string[] = content.value.split('\n');
+					if (!tags['workspace.py.any-azure']) {
+						tags['workspace.py.any-azure'] = WorkspaceStats.searchArray(dependencies, /azure/i);
+					}
+
+					// We're only interested in the '[packages]' section of the Pipfile
+					dependencies = dependencies.slice(dependencies.indexOf('[packages]') + 1, dependencies.indexOf('[dev-packages]'));
+
+					for (let dependency of dependencies) {
+						// All dependencies in Pipfiles follow the format: `<package> = <version, or git repo, or something else>`
+						let packageName = dependency.split('=')[0].trim();
+						if (PyModulesToLookFor.indexOf(packageName) > -1) {
```

Comment:
When the `dependency` here doesnt have the `=` in it, the `packageName` will be equal to `dependency`